### PR TITLE
[Merged by Bors] - Engine API v1.0.0.alpha.6 + interop tests

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

This PR extends #3018 to address my review comments there and add automated integration tests with Geth (and other implementations, in the future).

I've also de-duplicated the "unused port" logic by creating an  `common/unused_port` crate.

## Additional Info

I'm not sure if we want to merge this PR, or update #3018 and merge that. I don't mind, I'm primarily opening this PR to make sure CI works.

### CODE REVIEW

File: beacon_node/execution_layer/src/lib.rs [hunk 1]
```diff
@@ -577,13 +597,77 @@ impl ExecutionLayer {
...
+        for result in broadcast_results {
+            match result {
+                Ok(response) => match (&response.payload_status.latest_valid_hash, &response.payload_status.status) {
+                    (None, &PayloadStatusV1Status::Valid) => valid += 1,
+                    (Some(latest_hash), &PayloadStatusV1Status::Valid) => {
+                        if latest_hash == &head_block_hash {
+                            valid += 1;
+                        } else {
+                            // According to a strict interpretation of the spec, the EE should never
+                            // respond with `VALID` *and* a `latest_valid_hash`.
```
Comment (ethDreamer) [inline beacon_node/execution_layer/src/lib.rs]: This is not true, the [spec says](https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#specification-1) it **MUST** return

`{payloadStatus: {status: VALID, latestValidHash: forkchoiceState.headBlockHash, validationError: null}, payloadId: null}`

Comment (ethDreamer) [reply beacon_node/execution_layer/src/lib.rs]: the comment is in the wrong case, the spec says it should never return

`{payloadStatus: {status: VALID, latestValidHash: null, ... }` - your code is accepting that case, not sure if we should be though..

Comment (paulhauner) [reply beacon_node/execution_layer/src/lib.rs]: Yeah, you're totally right. I'm dealing with that over here too: https://github.com/ethereum/go-ethereum/issues/24404

I'll tidy this up :pray:

**Comment by paulhauner** [comment]
I've just bumped `cc` to 1.0.73 as per https://github.com/alexcrichton/cmake-rs/issues/141.

I think this break is because Github Actions has migrated to Windows 2022 for runners.

**Comment by paulhauner** [comment]
Turns out there are more issues with Windows 2022. I'm going to drop CI back to Windows 2019 for the time being. I opened an issue at #3028.

**Comment by paulhauner** [comment]
Seems like the latest windows failure is just a timeout :roll_eyes: 

bors r+

**Comment by paulhauner** [comment]
Windows is timing out before the execution engines come online, failing the new execution engine integration tests. I've skipped the integration tests for Windows for the time being.

This is tracked at https://github.com/sigp/lighthouse/issues/3030

**Pull request closed without merging**